tools/ddprobe.cpp

```diff
@@ -69,9 +69,131 @@ syncThreadDesktop() {
+ *
+ * @param dup A reference to the DXGI output duplication object (`dxgi::dup_t&`) used to acquire frames.
+ * @param device A pointer to the ID3D11Device interface that represents the device associated with the Direct3D context.
+ * @return HRESULT Returns `S_OK` if a non-black frame is captured successfully, `S_FALSE` if all frames are black, or an error code if a failure occurs during the process.
+ *
+ * Possible return values:
+ * - `S_OK`: A non-black frame was captured, indicating capture was successful.
+ * - `S_FALSE`: All 10 frames were black, indicating potential capture issues.
+ * - `E_FAIL`, `DXGI_ERROR_*`, or other DirectX HRESULT error codes in case of specific failures during frame capture, texture creation, or resource mapping.
+ */
+HRESULT
+test_frame_capture(dxgi::dup_t &dup, ID3D11Device *device) {
+  for (int i = 0; i < 10; ++i) {
+    std::cout << "Attempting to acquire the next frame (" << i + 1 << "/10)..." << std::endl;
+    IDXGIResource *frameResource = nullptr;
+    DXGI_OUTDUPL_FRAME_INFO frameInfo;
+    HRESULT status = dup->AcquireNextFrame(500, &frameInfo, &frameResource);
+    if (FAILED(status)) {
+      std::cout << "Failed to acquire next frame [0x" << std::hex << status << "]" << std::endl;
+      return status;
+    }
+
+    std::cout << "Frame acquired successfully." << std::endl;
+
+    ID3D11Texture2D *frameTexture = nullptr;
+    status = frameResource->QueryInterface(__uuidof(ID3D11Texture2D), (void **) &frameTexture);
+    frameResource->Release();
+    if (FAILED(status)) {
+      std::cout << "Failed to query texture interface from frame resource [0x" << std::hex << status << "]" << std::endl;
+      return status;
+    }
+
+    D3D11_TEXTURE2D_DESC frameDesc;
+    frameTexture->GetDesc(&frameDesc);
+    frameDesc.Usage = D3D11_USAGE_STAGING;
+    frameDesc.CPUAccessFlags = D3D11_CPU_ACCESS_READ;
+    frameDesc.BindFlags = 0;
+    frameDesc.MiscFlags = 0;
+
+    ID3D11Texture2D *stagingTexture = nullptr;
+    status = device->CreateTexture2D(&frameDesc, nullptr, &stagingTexture);
+    if (FAILED(status)) {
+      std::cout << "Failed to create staging texture [0x" << std::hex << status << "]" << std::endl;
+      frameTexture->Release();
+      return status;
+    }
+
+    ID3D11DeviceContext *context = nullptr;
+    device->GetImmediateContext(&context);
+    context->CopyResource(stagingTexture, frameTexture);
+    frameTexture->Release();
+
+    D3D11_MAPPED_SUBRESOURCE mappedResource;
+    status = context->Map(stagingTexture, 0, D3D11_MAP_READ, 0, &mappedResource);
+    if (SUCCEEDED(status)) {
```

Comment:
This conditional could potentially be reversed (or otherwise simplified) if RAII-based cleanup is implemented.

Reply:
I’ve fixed all that by using ComPtr, also just a side note… I don’t like negating Boolean checks. 

example…

if(!isEven) { …} ❌

vs if(is odd){ …} ✅

To me it is more clear to keep checks always test for truths, but that is a completely personal preference in coding styles on my part.